actors/market/tests/market_actor_test.rs

```diff
@@ -235,195 +234,136 @@ fn account_actor_check() {
+    let amount = TokenAmount::from(20);
+    add_provider_funds(&mut rt, amount.clone(), provider_addr, owner_addr, worker_addr);
 
     assert_eq!(amount, get_escrow_balance(&rt, &provider_addr).unwrap());
 
-    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, worker_addr);
-    expect_provider_control_address(&mut rt, provider_addr, owner_addr, worker_addr);
-
-    let withdraw_amount = TokenAmount::from(1u8);
-
-    rt.expect_send(
-        owner_addr,
-        METHOD_SEND,
-        RawBytes::default(),
+    // worker calls WithdrawBalance, balance is transferred to owner
+    let withdraw_amount = TokenAmount::from(1);
+    withdraw_provider_balance(
+        &mut rt,
         withdraw_amount.clone(),
-        RawBytes::default(),
-        ExitCode::OK,
+        withdraw_amount,
+        provider_addr,
+        owner_addr,
+        worker_addr,
     );
 
-    let params =
-        WithdrawBalanceParams { provider_or_client: provider_addr, amount: withdraw_amount };
-
-    assert!(rt
-        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap(),)
-        .is_ok());
-
-    rt.verify();
-
-    assert_eq!(get_escrow_balance(&rt, &provider_addr).unwrap(), TokenAmount::from(19u8));
+    assert_eq!(TokenAmount::from(19), get_escrow_balance(&rt, &provider_addr).unwrap());
+    // TODO: actor.checkState(rt)
 }
 
-#[ignore]
 #[test]
-fn withdraw_non_provider() {
-    // Test is currently failing because curr_epoch  is 0. When subtracted by 1, it goe snmegative causing a overflow error
+fn withdraws_from_non_provider_escrow_funds() {
     let mut rt = setup();
 
     let client_addr = Address::new_id(CLIENT_ID);
 
-    let amount = TokenAmount::from(20u8);
+    let amount = TokenAmount::from(20);
     add_participant_funds(&mut rt, client_addr, amount.clone());
 
-    assert_eq!(amount, get_escrow_balance(&rt, &client_addr).unwrap());
-
-    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, client_addr);
-    rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
-
-    let withdraw_amount = TokenAmount::from(1u8);
-
-    rt.expect_send(
-        client_addr,
-        METHOD_SEND,
-        RawBytes::default(),
-        withdraw_amount.clone(),
-        RawBytes::default(),
-        ExitCode::OK,
-    );
-
-    let params = WithdrawBalanceParams { provider_or_client: client_addr, amount: withdraw_amount };
-
-    assert!(rt
-        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap(),)
-        .is_ok());
+    assert_eq!(get_escrow_balance(&rt, &client_addr).unwrap(), amount);
 
-    rt.verify();
+    let withdraw_amount = TokenAmount::from(1);
+    withdraw_client_balance(&mut rt, withdraw_amount.clone(), withdraw_amount, client_addr);
 
-    assert_eq!(get_escrow_balance(&rt, &client_addr).unwrap(), TokenAmount::from(19u8));
+    add_participant_funds(&mut rt, client_addr, amount);
```

Comment:
this looks like a typo, to complete the test assert that the escrow balance at the client_addr is now 19.

Reply:
Yes that was a typo. Thanks.